rust/operator-binary/src/controller.rs

```diff
@@ -945,6 +952,19 @@ fn build_recommended_labels<'a>(
     }
 }
 
+/// Returns our semver representation for product config e.g. 377.0.0
+pub fn trino_version_to_semver(image_version: &str) -> Result<String, Error> {
```

Comment:
I would rather call it this way as we pass in `resolved_product_image.product_version`
```suggestion
pub fn trino_version_to_semver(product_version: &str) -> Result<String, Error> {
```

You are aware that  `resolved_product_image.product_version` produces something like `396`? I think we  don't need the split() here.
FYI: There are special functions if you only need the first part of the contents after split(). Performance and so ;)